crates/autopilot/src/domain/cow_amm.rs

```diff
@@ -0,0 +1,33 @@
+use {
+    cow_amm::Amm,
+    primitive_types::H160,
+    std::{collections::HashSet, sync::Arc},
+};
+
+/// Cow AMM
+/// The list of owners is initialized with the values specified in the autopilot
+/// configuration file, and it is extended with all the CoW AMM owners which are
+/// deployed by the CoW AMM helper contract
+#[derive(Clone)]
+pub struct CowAmm {
+    cow_amm_registry: cow_amm::Registry,
+}
+
+impl CowAmm {
+    pub fn new(cow_amm_registry: cow_amm::Registry) -> Self {
+        Self { cow_amm_registry }
+    }
+
+    pub async fn list_addresses(&self) -> HashSet<H160> {
+        self.cow_amm_registry
+            .cow_amms()
+            .await
+            .into_iter()
+            .map(|cow_amm| *cow_amm.address())
+            .collect::<HashSet<_>>()
+    }
+
+    pub async fn list_cow_amms(&self) -> Vec<Arc<Amm>> {
+        self.cow_amm_registry.cow_amms().await
+    }
```

Comment:
Can we just extend the registry by adding the `list_addresses` function? This struct won't be required as well as the `list_cow_amms` `function`.

Reply:
I would also prefer to consider the `Registry` type as the `domain` type since its interface is already pretty reasonable.
Overall I feel like this code doesn't offer enough value to be worth this wrapper type.

Reply:
One potential benefit of having a wrapper would be to enforce `list_cow_amms` to always return same values for a single `Auction` (by keeping internal cached dictionary for each auction). This obviously can be achieved in other ways (like making `Vec<Arc<Amm>>` part of the `AuctionWithId` which means is determined only once the Auction is built).

Bottom like, do not make `cow_amm_registry` available to `RunLoop` as there are at least two places (for now) where we need `Vec<Arc<Amm>>` (one when we send solve request to drivers, second when we save the cow amms to db), so calling `cow_amm_registry.cow_amms().await` multiple times for the same Auction might result in bug.

Reply:
Having the list of `Amm` cached per auction can backfire very badly with OOM errors. And the maintenance around it for deleting the already used auctions and so on can be very cumbersome. I think this approach is way more  bug prone that calling twice `cow_amm_registry.cow_amms().await` especially when new amms are not expected to be indexed often. But I agree calling it in two different places is sub-optimal and can lead to bugs.

Reply:
I think we can simply save the cow amms for last auction build (we do not actually need to save history). Anyway, this is just a small improvement to make it harder for users of this struct to make mistake.